extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js

```diff
@@ -368,27 +372,66 @@ export class LightboxManager {
    */
   getThumbnails(lightboxGroupId) {
     return this.lightboxGroups_[lightboxGroupId]
-        .map((element, i) => ({
-          url: this.getThumbnailUrl_(dev().assertElement(element), i),
+        .map(element => ({
+          srcset: this.getThumbnailSrcset_(dev().assertElement(element)),
+          placeholderUrl: this.getPlaceholderForElementType_(element),
           element,
         }));
   }
 
   /**
-   * Get thumbnail url for single element.
+   * Returns the default placeholder based on element type
    * @param {!Element} element
-   * @param {number=} index fake it for testing only, will delete later
    * @return {string}
    * @private
    */
-  getThumbnailUrl_(element, index) {
-    if (element.hasAttribute('lightbox-thumbnail-src')) {
-      return element.getAttribute('lightbox-thumbnail-src');
+  getPlaceholderForElementType_(element) {
+    switch (element.tagName) {
```

Comment:
nit: Closure might automagically convert this already, but this should be simplified to a single return statement until the switch syntax is actually used when #12713 is implemented.